RoboJackets/robocup-software
# Make GitHub Actions Auto-style Correctly

## Description

This PR improves upon #2015 by making automatic style formatting happen in a separate PR, which tags the original PR author to review and merge it. That way, on merge, the other automated checks will run after style fixes are automatically applied. (Currently the auto-style format bricks other checks from running.)

See the very last commits and #2028 for usage example. (Sorry for the terrible commit messages.)

## Steps to Test
### Test Case 1
1. Pull this branch
2. Make a style error (e.g. add an extra space to mess up indentation)
3. Wait a few minutes, see if the auto-fix PR pops up and tags me (due to GH limitations I can only set it to tag the PR author)

**Expected result:** A PR by GitHub Actions bot should be created, tagging both this PR and its author (me)

### Test Case 2
1. After Test Case 1, merge the auto-fix PR in for me ("squash and merge")

**Expected result:** The style fixes will be applied, and build-and-test should re-run and appear at the bottom of this PR. (This can take a while.)
